ewatercycle/models/lisflood.py

```diff
@@ -0,0 +1,398 @@
+import os
+import subprocess
+import time
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from datetime import datetime
+from os import PathLike
+from pathlib import Path
+from typing import Any, Iterable, Optional, Tuple, Union
+
+import numpy as np
+import xarray as xr
+from cftime import num2date
+from grpc4bmi.bmi_client_docker import BmiClientDocker
+from grpc4bmi.bmi_client_singularity import BmiClientSingularity
+
+from ewatercycle import CFG
+from ewatercycle.forcing.forcing_data import ForcingData
+from ewatercycle.models.abstract import AbstractModel
+from ewatercycle.parametersetdb.config import AbstractConfig
+
+
+@dataclass
+class LisfloodParameterSet:
+    """Input files specific for parameterset, model boundaries, and configuration template files
+
+    Example:
+
+    .. code-block::
+
+        parameterset = LisfloodParameterSet(
+            root=Path('/projects/0/wtrcycle/comparison/lisflood_input/Lisflood01degree_masked'),
+            mask=Path('/projects/0/wtrcycle/comparison/recipes_auxiliary_datasets/LISFLOOD/model_mask.nc'),
+            config_template=Path('/projects/0/wtrcycle/comparison/lisflood_input/settings_templates/settings_lisflood.xml'),
+            lisvap_config_template=Path('/projects/0/wtrcycle/comparison/lisflood_input/settings_templates/settings_lisvap.xml'),
+        )
+    """
+    root: PathLike
+    """Directory with input files"""
+    mask: Path
+    """A NetCDF file with model boundaries"""
+    config_template: PathLike
+    """Config file used as template for a lisflood run"""
+    lisvap_config_template: Optional[PathLike] = None
+    """Config file used as template for a lisvap run"""
+
+
+# LISFLOOD settings file reference maps prefixes
+# MapsName: {prefix_name, prefix}
+MAPS_PREFIXES = {
```

Comment:
Not sure I understand what this is needed for.

Reply:
An explanation is added.